Приложение Playlist Maker

IvanPodlesnykh opened the pull request:
Реализовал главный экран с кнопками перехода на прочие экраны Реализовал экран настроек с возможностью возврата на главный экран Цвета, строки и повторяющиеся размеры вынесены в ресурсы Реализовал возврат с экрана настроек на главный экран через this.finish(), т.к. через startActivity(...) кидает новый главный экран наверх стека

IvanPodlesnykh committed:
Приложение Playlist Maker
Реализовал главный экран с кнопками перехода на прочие экраны
Реализовал экран настроек с возможностью возврата на главный экран
Цвета, строки и повторяющиеся размеры вынесены в ресурсы
Реализовал возврат с экрана настроек на главный экран через this.finish(), т.к. через startActivity(...) кидает новый гланый экран наверх стека

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,9 @@
+    <dimen name="main_padding_horizontal">16dp</dimen>
+    <dimen name="main_padding_bottom">28dp</dimen>
+    <dimen name="icon_padding">8dp</dimen>
+    <dimen name="button_corner_radius">15dp</dimen>
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,64 @@
+        app:iconGravity="textStart"
+        app:iconPadding="@dimen/icon_padding"
+        app:cornerRadius="@dimen/button_corner_radius"
+        android:backgroundTint="@color/white"></Button>
```
Лучше закрывать так: `/>`, если это не вьюгруппа c вьюхами внутри (типа LinearLayout)

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,107 @@
+            android:layout_gravity="center"/>
+
+        <TextView
+            android:layout_width="208dp"
```
wrap_content

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,107 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+            <TextView
+                android:layout_width="300dp"
```
Не стоит задавать конкретную ширину. Телефоны могут сильно отличаться по размерам, на разных телефонах будет выглядеть по разному. На очень маленьких или больших устройствах может смотреться плохо или даже что-то не влезть

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,107 @@
+            android:layout_height="wrap_content">
+            <TextView
+                android:layout_width="300dp"
+                android:layout_height="61dp"
```
Повторяющийся аналогичный размер, в ресурсы

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,107 @@
+            <TextView
+                android:layout_width="300dp"
+                android:layout_height="61dp"
+                android:paddingStart="16dp"
```
1) Повторяющийся аналогичный размер, в ресурсы
2) Лучше не прописывать отступы в каждом элементе, а задать общий отступ в родительском лейауте

IvanPodlesnykh committed:
Приложение Playlist Maker
1. Исправил текст кнопок на главном экране (был капсом)
2. Внес повторяющиеся размеры в dimens.xml

IvanPodlesnykh committed:
Приложение Playlist Maker
1. Исправил текст кнопок на главном экране (был капсом)
2. Внес повторяющиеся размеры в dimens.xml
